File: integration-test/src/test/java/com/promise/integrationtest/idpool/dto/AllocateIPv4Response.java
package com.promise.integrationtest.idpool.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class AllocateIPv4Response
{

    @JsonProperty(value = "Address", required = true)
    private String address;
    @JsonProperty(value = "Pool", required = true)
    private GetIPv4PoolResponse pool;

    public String getAddress()
    {
        return address;
    }

    public void setAddress(String address)
    {
        this.address = address;
    }

    public GetIPv4PoolResponse getPool()
    {
        return pool;
    }

    public void setPool(GetIPv4PoolResponse pool)
    {
        this.pool = pool;
    }
}

File: pool/object/dto/AllocateIPv4Response.go
package dto

// AllocateIPv4Response is the DTO to allocate an IP.
type AllocateIPv4Response struct {
	Address string
	Pool    GetIPv4PoolResponse
}

// String return the debug info.
func (dto AllocateIPv4Response) String() string {
	return dto.Address
}
